scripts/training/start.sh

```diff
@@ -69,11 +69,40 @@ if [ "$DR_WORKERS" -gt 1 ]; then
     COMPOSE_FILES="$COMPOSE_FILES $DR_DOCKER_FILE_SEP $DR_DIR/docker/docker-compose-robomaker-multi.yml"
   fi
 
-  export ROBOMAKER_COMMAND="./run.sh multi distributed_training.launch"
+  if [ "$DR_MULTI_CONFIG" == "True" ]; then
+    echo "Multi-config training"
+
+    i=1
+    while [[ $i -le $DR_WORKERS ]] 
+    do
+        YAML_ARRAY[$i]=$(echo ${DR_LOCAL_S3_TRAINING_PARAMS_FILE} | sed s/.yaml/_$i.yaml/)
+        declare "MT_S3_TRAINING_PARAMS_FILE_$i=${YAML_ARRAY[$i]}"
+        ((i = i + 1))
+    done
+
+    export MT_S3_TRAINING_PARAMS_FILE_1=$MT_S3_TRAINING_PARAMS_FILE_1
+    export MT_S3_TRAINING_PARAMS_FILE_2=$MT_S3_TRAINING_PARAMS_FILE_2
+    export MT_S3_TRAINING_PARAMS_FILE_3=$MT_S3_TRAINING_PARAMS_FILE_3
+    export MT_S3_TRAINING_PARAMS_FILE_4=$MT_S3_TRAINING_PARAMS_FILE_4
+    export MT_S3_TRAINING_PARAMS_FILE_5=$MT_S3_TRAINING_PARAMS_FILE_5
+    export MT_S3_TRAINING_PARAMS_FILE_6=$MT_S3_TRAINING_PARAMS_FILE_6
+    export MT_S3_TRAINING_PARAMS_FILE_7=$MT_S3_TRAINING_PARAMS_FILE_7
+    export MT_S3_TRAINING_PARAMS_FILE_8=$MT_S3_TRAINING_PARAMS_FILE_8
+
+    # read in multiconfig.txt file, and export the world files
+    source multiconfig.txt 
+
+    # this command tells the robomaker worker which world_name and params_file to use
+    export ROBOMAKER_COMMAND='if [[ "$DOCKER_REPLICA_SLOT" == *"1"* ]]; then export WORLD_NAME=$DR_MT_WORLD_NAME_1; export S3_YAML_NAME=$MT_S3_TRAINING_PARAMS_FILE_1; elif [[ "$DOCKER_REPLICA_SLOT" == *"2"* ]]; then export WORLD_NAME=$DR_MT_WORLD_NAME_2; export S3_YAML_NAME=$MT_S3_TRAINING_PARAMS_FILE_2; elif [[ "$DOCKER_REPLICA_SLOT" == *"3"* ]]; then export WORLD_NAME=$DR_MT_WORLD_NAME_3; export S3_YAML_NAME=$MT_S3_TRAINING_PARAMS_FILE_3; elif [[ "$DOCKER_REPLICA_SLOT" == *"4"* ]]; then export WORLD_NAME=$DR_MT_WORLD_NAME_4; export S3_YAML_NAME=$MT_S3_TRAINING_PARAMS_FILE_4; elif [[ "$DOCKER_REPLICA_SLOT" == *"5"* ]]; then export WORLD_NAME=$DR_MT_WORLD_NAME_5; export S3_YAML_NAME=$MT_S3_TRAINING_PARAMS_FILE_5; elif [[ "$DOCKER_REPLICA_SLOT" == *"6"* ]]; then export WORLD_NAME=$DR_MT_WORLD_NAME_6; export S3_YAML_NAME=$MT_S3_TRAINING_PARAMS_FILE_6; elif [[ "$DOCKER_REPLICA_SLOT" == *"7"* ]]; then export WORLD_NAME=$DR_MT_WORLD_NAME_7; export S3_YAML_NAME=$MT_S3_TRAINING_PARAMS_FILE_7; elif [[ "$DOCKER_REPLICA_SLOT" == *"8"* ]]; then export WORLD_NAME=$DR_MT_WORLD_NAME_8; export S3_YAML_NAME=$MT_S3_TRAINING_PARAMS_FILE_8; fi && ./run.sh multi distributed_training.launch'
```

Comment:
Let's have a look as to how we can combine this into something slightly more elegant. What we want to do is to pass in the WORLD_NAME and S3_YAML_NAME per worker, right? We can adapt the run.sh on the robomaker side to help on this too!

I would do the following; create a yaml or json and pass as separate (single) environment variable. We then adapt run.sh to push out its file based on the established WORKER_NUM which run.sh anyways needs to figure out.